File: app/src/main/java/com/thejoyrun/aptpreferences/MyApplication.java
package com.thejoyrun.aptpreferences;

import android.app.Application;

import com.alibaba.fastjson.JSON;

/**
 * Created by Wiki on 16/7/15.
 */

public class MyApplication extends Application{
    @Override
    public void onCreate() {
        super.onCreate();
        AptPreferencesManager.init(this, new AptParser() {

            @Override
            public Object deserialize(Class clazz, String text) {
                return JSON.parseObject(text,clazz);
            }

            @Override
            public String serialize(Object object) {
                return JSON.toJSONString(object);
            }
        });
    }
}

File: app-kotlin/src/main/java/com/thejoyrun/app_kotlin/MyApplication.kt
package com.thejoyrun.app_kotlin

import android.app.Application

import com.alibaba.fastjson.JSON
import com.thejoyrun.aptpreferences.AptParser
import com.thejoyrun.aptpreferences.AptPreferencesManager

/**
 * Created by Wiki on 16/7/15.
 */

class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        AptPreferencesManager.init(this, object : AptParser {

            override fun deserialize(clazz: Class<*>, text: String): Any? {
                return JSON.parseObject(text, clazz)
            }

            override fun serialize(`object`: Any): String {
                return JSON.toJSONString(`object`)
            }
        })
        AptPreferencesManager.setUserInfo("123456")
    }
}
